### ISSUE TITLE
Update project dependencies

### ISSUE THREAD

**Issue by sskjames** [mcruncher/worshipsongs-android#585]

**Comment by sskjames** [comment]
Got this error while trying to upgrade Material from 1.4.0 to 1.6.1.
```
 AAPT: error: resource android:color/system_neutral1_1000 not found.
```

**Comment by sskjames** [comment]
Got this error while trying to upgrade appcompat from 1.3.1 to 1.5.1.
```
AAPT: error: resource android:attr/lStar not found
```

**Comment by sskjames** [comment]
No test was executed when trying to upgrade androidx.preference from 1.1.1 to 1.2.0.

**Comment by madasamys** [comment]
Hi @sskjames upgrading minimum SDK version to 22 may not compatible for lower version (>22) running on device. 
I could see development activity in this repository on last month. I really appreciate you take time to contribute this project. It would be really great, release new version every quarter(4 months).

**Comment by sskjames** [comment]
Hi @madasamys, thank you very much for taking this project this far. We will try our best to release new features as suggested by you.

**Comment by sskjames** [comment]
Hi @madasamys, SDK 22 corresponds to Lollipop and since that's quite old, we need to move on.

**Comment by sskjames** [comment]
Resuming work on this after a very long time. Ran the unit tests and observed that only 42 tests were run.
However, in master, I could run 108 tests.

**Comment by sskjames** [comment]
Observed that the Spock tests are not running. Also noted that we've commented out the groovyx.android plugin.
When I enable this plugin, I get the following error:
```
A problem occurred evaluating settings 'worshipsongs-android'.
> Could not find method dependencyResolutionManagement() for arguments [settings_boxk098zlg4jjysw7107h9zxm$_run_closure1@669368cd] on settings 'worshipsongs-android' of type org.gradle.initialization.DefaultSettings.

```

**Comment by sskjames** [comment]
Looks like groovyx.android plugin is not compatible with recent versions of gradle and Android. What is even more shocking is that they have deprecated this, which means we cannot write Spock tests 😢.
Ref: [Groovy Android Plugin project](https://github.com/groovy/groovy-android-gradle-plugin)

**Comment by sskjames** [comment]
Removed the dependency roboelectric-shadows-support as Roboelectric has [removed support](https://robolectric.org/using-add-on-modules/) for this.

**Comment by sskjames** [comment]
After removing the groovyx.android plugin, we are not able to run Spock tests.
As of now, we are able to run plain JUnit and Roboelectric tests.
Totally 42 tests are run.

We are facing a number of difficulties in using Spock:
- We are not able to use the latest version of Spock because Roboelectric depends on JUnit 4.
- Not able to run Spock tests as groovyx.anroid plugin is not compatible with Gradle 7.

In view of this, it would be better to keep things simple and move away from Spock. Just stick to JUnit. As much as I love using Spock, I also want to keep this project in good maintainable state and that means as little dependencies as possible.

**Comment by sskjames** [comment]
Did a quick smoke test in Android 5 (API 22) and Android 7.1 (API 25).

**Comment by sskjames** [comment]
Tested in API 31 (Android 12).

**Comment by sskjames** [comment]
Tested in API 30 (Android 11).

**Comment by sskjames** [comment]
Tested in API 29 (Android 10).

**Comment by sskjames** [comment]
Tested in API 27 (Android 8.1) and API 28 (Android 9.0).

**Comment by sskjames** [comment]
Jenkins build fails with the following error:
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:testDebugUnitTest'.
> Failed to find target with hash string 'android-31' in: /var/jenkins_home/tools/android-sdk

```

**Comment by sskjames** [comment]
Checked with @mcr-paulanand and found that we don't have android-31 in our Jenkins master. 
Once we have it, we'll try building this again.

**Comment by sskjames** [comment]
It looks like the sdk and other required tools can be automatically downloaded if we use v2.2.0 of Android Gradle Plugin.
Ref: https://github.com/JakeWharton/sdk-manager-plugin

**Comment by sskjames** [comment]
We've installed Android 31 SDK in Jenkins master using Android command line build tools for Linux.
In future, we'd look at other options.
This helped the build to pass but the package couldn't be published to Play Store.
Need to look into that.

**Comment by sskjames** [comment]
Here's the error message:
`   > SigningConfig "releaseConfig" is missing required property "storePassword".`

The reason is, we've moved away from a classic Jenkins job to the type `Organization Folder` where jobs get created automatically for every branch and PR in the repo. 
In the old job, we have defined some env variables, which are not available here.

**Comment by sskjames** [comment]
I also had to update the build number in the master job as we have now moved to GitHub organization folder. Build was successful and published by Jenkins to Playstore as 3.6.0.452. It's available for open testing.
